extensions/healthie/actions/__tests__/sendFormCompletionRequest.ts

```diff
@@ -19,21 +54,173 @@ describe('sendFormCompletionRequest action', () => {
+              ...samplePayload.fields,
+              is_recurring: true,
+              frequency: 'daily',
+              ...(value as any),
+            },
+          },
+          onComplete,
+          onError
+        )
+
+        expect(onComplete).toHaveBeenCalled()
+        expect(
+          mockGetSdkReturn.createFormCompletionRequest
+        ).toHaveBeenCalledWith({
+          input: {
+            ...sampleFormCompletion,
+            is_recurring: true,
+            frequency: 'Daily',
+            hour: String(value.hour),
+            minute: String(value.minute),
+            period: value.period.toUpperCase(),
+          },
+        })
+      })
+    })
+
+    describe('Frequency = weekly', () => {
+      test.each([
+        {
+          weekday: 'Monday',
+          weekdayActual: 'Monday',
+        },
+        {
+          weekday: 'tuesday',
+          weekdayActual: 'Tuesday',
+        },
+        {
+          weekday: 'wednesdaY',
+          weekdayActual: 'Wednesday',
+        },
+      ])('Should call onComplete when %p', async (value) => {
+        await sendFormCompletionRequest.onActivityCreated(
+          {
+            ...samplePayload,
+            fields: {
+              ...samplePayload.fields,
+              is_recurring: true,
+              frequency: 'weekly',
+              hour: 1,
+              minute: 1,
+              period: 'PM',
+              weekday: value.weekday,
+            },
+          },
+          onComplete,
+          onError
+        )
+
+        expect(
+          mockGetSdkReturn.createFormCompletionRequest
+        ).toHaveBeenCalledWith({
+          input: {
+            ...sampleFormCompletion,
+            is_recurring: true,
+            frequency: 'Weekly',
+            hour: '1',
+            minute: '1',
+            period: 'PM',
+            weekday: value.weekdayActual,
+          },
+        })
+        expect(onComplete).toHaveBeenCalled()
+      })
+    })
+
+    describe('Frequency = monthly', () => {
+      test('Should call onComplete when "monthday" is not empty', async () => {
+        await sendFormCompletionRequest.onActivityCreated(
+          {
+            ...samplePayload,
+            fields: {
+              ...samplePayload.fields,
+              is_recurring: true,
+              frequency: 'monthly',
+              hour: 1,
+              minute: 1,
+              period: 'PM',
+              weekday: 'monday',
+              monthday: '27th',
+            },
+          },
+          onComplete,
+          onError
+        )
+
+        expect(onComplete).toHaveBeenCalled()
+        expect(
+          mockGetSdkReturn.createFormCompletionRequest
+        ).toHaveBeenCalledWith({
+          input: {
+            ...sampleFormCompletion,
+            is_recurring: true,
+            frequency: 'Monthly',
+            hour: '1',
+            minute: '1',
+            period: 'PM',
+            weekday: 'Monday',
+            monthday: '27th',
+          },
```

Comment:
test the recurrence ending field here (see below, exact fields depends on the implementation we use)

Reply:
Sure, I'll the tests